src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+
+        String[] arrDates;
+
+        StringBuilder result = new StringBuilder().append("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+
+            int salary = 0;
+
+            for (String dates : data) {
+                arrDates = dates.split(" ");
+                LocalDate dateFromData = LocalDate.parse(arrDates[0], FORMATTER);
+                if (localDateFrom.compareTo(dateFromData) <= 0
+                        && localDateTo.compareTo(dateFromData) >= 0
+                        && name.equals(arrDates[1])) {
+                    salary += (Integer.parseInt(arrDates[2])
+                            * Integer.parseInt(arrDates[3]));
```

Comment:
![image](https://user-images.githubusercontent.com/81781479/183252486-0159753d-3258-4349-9515-329b1cb7674a.png)
Before you create PR with your homework, you need to go through the description of the task and checklist under the task, and correct all the points described there. The mentor will not check the work until the checklist items are corrected

Reply:
It is no longer possible to create a constant, the names of the fields are already informative, please clarify this comment